apps/openmw/mwworld/cellstore.cpp

```diff
@@ -179,8 +182,110 @@ namespace MWWorld
         return (ref.mRef.mRefnum == pRefnum);
     }
 
+    void CellStore::moveFrom(const Ptr &object, CellStore *from)
+    {
+        mHasState = true;
+        MovedRefTracker::iterator found = mMovedToAnotherCell.find(object.getBase());
+        if (found != mMovedToAnotherCell.end())
+        {
+            // A cell we had previously moved an object to is returning it to us.
+            assert (found->second == from);
+            mMovedToAnotherCell.erase(found);
+        }
+        else
+        {
+            mMovedHere.insert(std::make_pair(object.getBase(), from));
+        }
+
+        if (mState == State_Loaded)
```

Comment:
This has potentially poor performance when moving many actors between cells. The whole point of the preload state is to not load cells that we actually don't need to touch. But that is clearly not the case here. I suggest to just load the cell first when in preload state instead of the else branch.